chips/sam4l/src/adc.rs

```diff
@@ -408,12 +447,19 @@ impl hil::adc::Adc for Adc {
             self.active.set(true);
             self.continuous.set(true);
 
+            let trgsel;
+            if self.cpu_clock.get() {
+                trgsel = 1; // internal timer trigger
+            } else {
+                trgsel = 3; // continuous mode
```

Comment:
Is continuous mode actually correct here? Per the SAM4L manual, continuous mode means that a conversion is triggered at each clock cycle of the ADC clock. However, conversions take multiple cycles of the ADC clock, depending on various settings (see page 1000, Table 38-2) and additional triggers while a conversion is in progress are ignored. My guess is that based on your current setup, samples will actually occur at 1/7th the frequency you're expecting. A good way to test this is to toggle a GPIO pin within the ADC interrupt.